src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -169,13 +199,24 @@ DxcCompilationResult dxcCompile(const CHLSLCompiler* compiler, nbl::asset::hlsl:
 
 std::string CHLSLCompiler::preprocessShader(std::string&& code, IShader::E_SHADER_STAGE& stage, const SPreprocessorOptions& preprocessOptions) const
 {
+    std::ostringstream insertion;
+    insertion << IShaderCompiler::PREPROC_DIRECTIVE_ENABLER;
+    insertion << "line 1\n";
+    insertIntoStart(code, std::move(insertion));
```

Comment:
just use [std::string::insert](https://cplusplus.com/reference/string/string/insert/) in a single liner